omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java

```diff
@@ -258,15 +261,53 @@ public static void setFullySpecifiedName(Concept instance, String name) {
 	 * It's needed, because of ConversionException: Don't know how to handle collection class:
 	 * interface java.util.Collection
 	 * 
+	 * if request to update Concept updates ConceptName, adequate resource takes care of it,
+	 * so this method just adds new and removes deleted names.
+	 * 
 	 * @param instance
-	 * @param names
+	 * @param newNames
 	 */
 	@PropertySetter("names")
-	public static void setNames(Concept instance, List<ConceptName> names) {
-		for(ConceptName name : names){
+	public static void setNames(Concept instance, List<ConceptName> newNames) {
+		for(ConceptName name : newNames){
 			name.setConcept(instance);
 		}
-		instance.setNames(new HashSet<ConceptName>(names));
+		//delete names which are absent in list of new names
+		for (ConceptName name : instance.getNames()){
+			if(!checkIfNamesContainNameByProperties(newNames, name)){
+				instance.removeName(name);
+			}
+		}
+		//add names which are absent in list of old names
+		for(ConceptName name : newNames){
+			if(!checkIfNamesContainNameByProperties(instance.getNames(), name)){
+				instance.addName(name);
+			}
+		}
+	}
+	/**
+	 * helper method for {@link #setNames(Concept, List<ConceptName>)}
+	 * checks if specified name is present in concept's names list
+	 * 
+	 * @should return true if list contains name with given uuid
+	 * @should return true if list contains name with same properties
+	 * @should return false if there is no concept with same properties
+	 * 
+	 * @param names
+	 * @param searched
+	 * @return
+	 */
+	public static boolean checkIfNamesContainNameByProperties(Collection<ConceptName> names, ConceptName searched) {
+		for(ConceptName name : names){
+			if(Objects.equals(name.getUuid(), searched.getUuid())) return true;

```

Comment:
Our code convention is if clause statements should be inside curly braces and the statements should be on a new line